genesyscloud/routing_utilization/resource_routing_utilization_utils.go

```diff
@@ -0,0 +1,110 @@
+	settingsMap := settings[0].(map[string]interface{})
+
+	maxCapacity := settingsMap["maximum_capacity"].(int)
+	includeNonAcd := settingsMap["include_non_acd"].(bool)
+
+	// Optional
+	interruptableMediaTypes := &[]string{}
+	if types, ok := settingsMap["interruptible_media_types"]; ok {
+		interruptableMediaTypes = lists.SetToStringList(types.(*schema.Set))
+	}
+
+	return platformclientv2.Mediautilization{
+		MaximumCapacity:         &maxCapacity,
+		IncludeNonAcd:           &includeNonAcd,
+		InterruptableMediaTypes: interruptableMediaTypes,
+	}
+}
+
+func BuildLabelUtilizationsRequest(labelUtilizations []interface{}) map[string]LabelUtilization {
+	request := make(map[string]LabelUtilization)
+	for _, labelUtilization := range labelUtilizations {
+		labelUtilizationMap := labelUtilization.(map[string]interface{})
+		interruptingLabelIds := lists.SetToStringList(labelUtilizationMap["interrupting_label_ids"].(*schema.Set))
+
+		request[labelUtilizationMap["label_id"].(string)] = LabelUtilization{
+			MaximumCapacity:      int32(labelUtilizationMap["maximum_capacity"].(int)),
+			InterruptingLabelIds: *interruptingLabelIds,
+		}
+	}
+	return request
+}
+
+func FlattenUtilizationSetting(settings MediaUtilization) []interface{} {
+	settingsMap := make(map[string]interface{})
+
+	settingsMap["maximum_capacity"] = settings.MaximumCapacity
+	settingsMap["include_non_acd"] = settings.IncludeNonAcd
+	if settings.InterruptableMediaTypes != nil {
+		settingsMap["interruptible_media_types"] = lists.StringListToSet(settings.InterruptableMediaTypes)
+	}
+
+	return []interface{}{settingsMap}
+}
+
+func FilterAndFlattenLabelUtilizations(labelUtilizations map[string]LabelUtilization, originalLabelUtilizations []interface{}) []interface{} {
+	flattenedLabelUtilizations := make([]interface{}, 0)
+
+	for _, originalLabelUtilization := range originalLabelUtilizations {
+		originalLabelId := (originalLabelUtilization.(map[string]interface{}))["label_id"].(string)
+
+		for currentLabelId, currentLabelUtilization := range labelUtilizations {
+			if currentLabelId == originalLabelId {
+				flattenedLabelUtilizations = append(flattenedLabelUtilizations, flattenLabelUtilization(currentLabelId, currentLabelUtilization))
+				delete(labelUtilizations, currentLabelId)
+				break
+			}
+		}
+	}
+
+	return flattenedLabelUtilizations
+}
+
+func flattenLabelUtilization(labelId string, labelUtilization LabelUtilization) map[string]interface{} {
+	utilizationMap := make(map[string]interface{})
+
+	utilizationMap["label_id"] = labelId
+	utilizationMap["maximum_capacity"] = labelUtilization.MaximumCapacity
+	if labelUtilization.InterruptingLabelIds != nil {
+		utilizationMap["interrupting_label_ids"] = lists.StringListToSet(labelUtilization.InterruptingLabelIds)
+	}
+
+	return utilizationMap
+}
+
+// TODO: remove when routing skill group is refactored
+func buildHeaderParams(routingAPI *platformclientv2.RoutingApi) map[string]string {
```

Comment:
Can this be removed now that it's refactored?

Reply:
No. routing skill group not yet refactored